src/server.h

```diff
@@ -3336,15 +3340,19 @@ robj *setTypeDup(robj *o);
 #define HASH_SET_COPY 0
 
 
+void dbUpdateKeyWithVolaItemsTracking(serverDb *db, robj *o);
 void hashTypeFreeVolatileSet(robj *o);
-void hashTypeTrackEntry(robj *o, void *entry);
-void hashTypeUntrackEntry(robj *o, void *entry);
-void hashTypeTrackUpdateEntry(robj *o, void *old_entry, void *new_entry, long long old_expiry, long long new_expiry);
+void hashTypeTrackEntry(serverDb *db, robj *o, void *entry);
+void hashTypeUntrackEntry(serverDb *db, robj *o, void *entry);
+void hashTypeTrackUpdateEntry(serverDb *db, robj *o, void *old_entry, void *new_entry, long long old_expiry, long long new_expiry);
+vset *hashTypeGetVolatileSet(robj *o);
+size_t hashTypeReclaimExpiredFields(robj *o, serverDb *db, mstime_t now, unsigned long max_entries);
```

Comment:
This, again, is mixing 2 APIs.  Reclaiming expired fields - within a hashtype object - should be independent of the DB.  I recommend removing DB from the parameter list, and repositioning DB related functionality in `db.c`.